src/components/voicemail/VoicemailComponent.js

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import * as Flex from '@twilio/flex-ui';
+import moment from 'moment';
+import 'moment-timezone';
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import Tooltip from '@material-ui/core/Tooltip';
+import Icon from '@material-ui/core/Icon';
+
+import styles from './VoicemailStyles';
+import { inqueueUtils } from '../common';
+
+export default class VoicemailComponent extends React.Component {
+  static displayName = 'VoicemailComponent';
+
+  constructor(props) {
+    super(props);
+
+    this.state = {};
+  }
+
+  /*
+   * create outbound call from Flex using Actions API 'StartOutboundCall'
+   *
+   */
+  vmCallButtonAccessibility = async (state) => inqueueUtils(this.props.task, 'voicemail', state);
+
+  startTransfer = async () => inqueueUtils(this.props.task);
+
+  // web service call to delete the call recording/transcript
+  deleteResources = async () => inqueueUtils.deleteResource(this.props.task);
+
+  startCall = async () => {
+    const manager = Flex.Manager.getInstance();
+    const activityName = manager.workerClient.activity.name;
+    if (activityName === 'Offline') {
+      // eslint-disable-next-line no-alert
+      alert('Change activity state from "Offline" to place call to contact');
+      return;
+    }
+
+    await this.vmCallButtonAccessibility(true);
```

Comment:
Potentially related to the changes around this file, but this is the error I've been getting for voicemail:

```
Warning: Can't call setState (or forceUpdate) on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.
    in t (created by Styled(t))
    in Styled(t) (created by n)
    in div (created by Styled(div))
    in Styled(div) (created by n)
    in DynamicComponent (created by n)
    in n (created by WithTheme(n))
    in WithTheme(n) (created by Connect(WithTheme(n)))
    in Connect(WithTheme(n)) (created by NoTasksCanvas)
    in NoTasksCanvas (created by AgentDesktopView.Panel1)
window.console.error @ VM156:37
overrideMethod @ react_devtools_backend.js:2430
console.<computed> @ plugin-queued-callbacks-and-voicemail.js:5580
warningWithoutStack @ react-dom.development.js:506
warnAboutUpdateOnUnmounted @ react-dom.development.js:15753
scheduleWork @ react-dom.development.js:17050
enqueueSetState @ react-dom.development.js:11746
Component.setState @ react.development.js:501
n.onload @ twilio-flex.unbundled-react.min.js:1550
load (async)
t.validateImageUrl @ twilio-flex.unbundled-react.min.js:1550
t @ twilio-flex.unbundled-react.min.js:1550
constructClassInstance @ react-dom.development.js:11892
updateClassComponent @ react-dom.development.js:13614
beginWork @ react-dom.development.js:14213
performUnitOfWork @ react-dom.development.js:16539
workLoop @ react-dom.development.js:16577
renderRoot @ react-dom.development.js:16656
performWorkOnRoot @ react-dom.development.js:17510
performWork @ react-dom.development.js:17418
performSyncWork @ react-dom.development.js:17390
requestWork @ react-dom.development.js:17278
scheduleWork @ react-dom.development.js:17072
enqueueSetState @ react-dom.development.js:11746
Component.setState @ react.development.js:501
u.onStateChange @ twilio-flex.unbundled-react.min.js:163
notify @ twilio-flex.unbundled-react.min.js:163
t.notifyNestedSubs @ twilio-flex.unbundled-react.min.js:163
u.onStateChange @ twilio-flex.unbundled-react.min.js:163
notify @ twilio-flex.unbundled-react.min.js:163
t.notifyNestedSubs @ twilio-flex.unbundled-react.min.js:163
u.onStateChange @ twilio-flex.unbundled-react.min.js:163
notify @ twilio-flex.unbundled-react.min.js:163
t.notifyNestedSubs @ twilio-flex.unbundled-react.min.js:163
u.onStateChange @ twilio-flex.unbundled-react.min.js:163
notify @ twilio-flex.unbundled-react.min.js:163
t.notifyNestedSubs @ twilio-flex.unbundled-react.min.js:163
u.onStateChange @ twilio-flex.unbundled-react.min.js:163
p @ twilio-flex.unbundled-react.min.js:157
(anonymous) @ twilio-flex.unbundled-react.min.js:1620
(anonymous) @ twilio-flex.unbundled-react.min.js:163
e.dispatch @ twilio-flex.unbundled-react.min.js:152
e.dispatchWorkerAction @ twilio-flex.unbundled-react.min.js:169
e.dispatchAction @ twilio-flex.unbundled-react.min.js:169
handleReservationUpdated @ twilio-flex.unbundled-react.min.js:169
i @ twilio-flex.unbundled-react.min.js:169
n.emit @ twilio-flex.unbundled-react.min.js:1602
_emitEvent @ twilio-flex.unbundled-react.min.js:1602
_reservationCleanupEventsHandler @ twilio-flex.unbundled-react.min.js:1602
n.emit @ twilio-flex.unbundled-react.min.js:1602
webSocket.onmessage @ twilio-flex.unbundled-react.min.js:1609
Show 20 more frames
plugin-queued-callbacks-and-voicemail.js:53624 Uncaught (in promise) TypeError: Object(...) is not a function
    at _callee2$ (plugin-queued-callbacks-and-voicemail.js:53624)
```

Reply:
It is not :( this happens on `master` too. I haven't looked where this is happening but I think it's because of the way we have the components registered

Reply:
Actually, voicemail works in `main`. I tested it after our last conversation and didn't see any issues other than the counter being wrong.

Reply:
@bermudezmt I've got a different error but in the same file, which I fixed in 9802a16. Can you try again with the code in #29?